consensus/src/experimental/buffer_manager.rs

```diff
@@ -157,13 +163,97 @@ impl StateManager {
             persisting_phase_rx,
 
             block_rx,
-            reset_rx,
+            sync_rx,
             end_epoch: false,
 
             verifier,
         }
     }
 
+    async fn process_ordered_blocks(
+        &mut self,
+        ordered_blocks: OrderedBlocks,
+    ) -> anyhow::Result<()> {
+        let OrderedBlocks {
+            blocks,
+            ordered_proof,
+            callback,
+        } = ordered_blocks;
+
+        // push blocks to buffer
+        for eb in blocks.iter() {
+            self.buffer
+                .push_back(BufferItem::Block(Arc::new(eb.clone())));
+        }
+
+        self.buffer.push_back(BufferItem::Proof(Box::new(ProofItem(
+            ordered_proof.clone(),
+            BTreeMap::new(),
+            callback,
+        ))));
+
+        // send blocks to execution phase
+        self.execution_phase_tx
+            .send(ExecutionRequest {
+                blocks: blocks
+                    .iter()
+                    .map(|eb| eb.block().clone())
+                    .collect::<Vec<Block>>(),
+            })
+            .await?;
+        Ok(())
+    }
+
+    async fn process_sync_req(&mut self, sync_event: SyncRequest) -> anyhow::Result<()> {
+        let SyncRequest {
+            tx,
+            ledger_info,
+            reconfig,
+        } = sync_event;
+
+        if reconfig {
+            // buffer manager will stop
+            self.end_epoch = true;
+        } else {
+            // clear the buffer until (including) the ledger_info
+            while let Some(ptr) = self.buffer.pop_front() {
+                if let BufferItem::Proof(fp_item) = ptr {
```

Comment:
avoid using abbr like fp, better to use proof_item

Reply:
resolved